src/h5cpp/node/functions.hpp

```diff
@@ -153,11 +182,12 @@ DLL_EXPORT void remove(const Group &base,const Path &rel_path,
 //! \param target the target node (could also be path here?)
 //! \param link_base the base group for the link location
 //! \param link_path path to the new link (either relative or absolute)
-//! \param lcpl optional reference to a link creation property list
-//! \param lapl optional reference to a link access property list
+//! \param lcpl optional link creation property list
+//! \param lapl optional link access property list
 //!
 DLL_EXPORT void link(const Node &target,
```

Comment:
Here is something I am missing. We have a function to create a link to an existing target (target). However, we should also be able to create links to targets that do not yet exist. In other words we need an additional function where target is a target_path (like in the case for external links).

Reply:
We now have this. There are now 3 versions of the `link` function:
1) for creating an external link, path need not resolve to an existing node
2) soft link within file, path need not resolve to an existing node
3) link to node, with `Node` as parameter, regardless of file, picks the right one from above
All of these are available as free functions as well as member functions of group.
In the process I have discovered that some of the error checking is redundant and many edge cases were already taken care of by the underlying H5 routines. I eliminated some checks and tests still confirm the same behavior. There may be more such superfluous error checking in the other functions. We can review these at a later time. For now, we have the necessary functionality in place.